packages/alfa-rules/test/sia-r83/rule.spec.tsx

```diff
@@ -125,70 +130,74 @@ test(`evaluate() fails a text node that clips overflow by not wrapping text
 });
 
 test("evaluate() is inapplicable to a text node that is not visible", async (t) => {
-  const document = Document.of([
-    <div
-      style={{
-        overflow: "hidden",
-        whiteSpace: "nowrap",
-      }}
-      hidden
-    >
-      Hello world
-    </div>,
-  ]);
+  const document = Document.of(
+    [<div hidden>Hello world</div>],
+    [
+      h.sheet([
+        h.rule.style("div", {
+          overflow: "hidden",
+          whiteSpace: "nowrap",
+        }),
+      ]),
+    ]
+  );
 
   t.deepEqual(await evaluate(R83, { document }), [inapplicable(R83)]);
 });
 
 test(`evaluate() is inapplicable to a text node that is excluded from the
       accessibility tree using the \`aria-hidden\` attribute`, async (t) => {
-  const document = Document.of([
-    <div
-      style={{
-        overflow: "hidden",
-        whiteSpace: "nowrap",
-      }}
-      aria-hidden="true"
-    >
-      Hello world
-    </div>,
-  ]);
+  const document = Document.of(
+    [<div aria-hidden="true">Hello world</div>],
+    [
+      h.sheet([
+        h.rule.style("div", {
+          overflow: "hidden",
+          whiteSpace: "nowrap",
+        }),
+      ]),
+    ]
+  );
 
   t.deepEqual(await evaluate(R83, { document }), [inapplicable(R83)]);
 });
 
-test("evaluates() fails a clipped node with `normal` line-height", async (t) => {
-  const target = h.text("Hello World");
-
-  const document = Document.of([
-    <div
-      style={{
-        overflow: "hidden",
-        height: "1.5em",
-        fontSize: "16px",
-      }}
-    >
-      {target}
-    </div>,
-  ]);
+test(`evaluate() is inapplicable to a text node with a fixed absolute height set
+      via the style attribute`, async (t) => {
+  const document = Document.of(
+    [
+      <div
+        style={{
+          height: "20px",
+        }}
+      >
+        Hello world
+      </div>,
+    ],
+    [
+      h.sheet([
+        h.rule.style("div", {
+          overflow: "hidden",
+        }),
+      ]),
+    ]
+  );
 
-  t.deepEqual(await evaluate(R83, { document }), [
-    failed(R83, target, { 1: Outcomes.ClipsText }),
-  ]);
+  t.deepEqual(await evaluate(R83, { document }), [inapplicable(R83)]);
 });
 
-test(`evaluate() is inapplicable to a text node with a fixed width and hidden
-      overflow`, async (t) => {
-  const document = Document.of([
-    <div
-      style={{
-        overflow: "hidden",
-        width: "80ch",
-      }}
-    >
-      Hello world
-    </div>,
-  ]);
+test(`evaluate() is inapplicable to a text node with a fixed relative height`, async (t) => {
+  const document = Document.of(
+    [<div>Hello world</div>],
+    [
+      h.sheet([
+        h.rule.style("div", {
+          overflow: "hidden",
+          height: "1.2em",
```

Comment:
This is actually clipping the second line of text.
(Well, sort of, because there is top and bottom white space in the `<div>`, but depending on the height, this is clipped when there are several lines of text).
`height: 1.5em`:
<img width="182" alt="clip" src="https://user-images.githubusercontent.com/19969498/109811606-59fb2000-7c2b-11eb-87bf-1aebed5a6ea2.png">

Reply:
Yes and that case is immediately obvious without scaling the text as that will _always_ cause clipping. Without a layout system, however, we can't accurately call out these cases without a high risk of false positives. The rule will therefore only call out the less obvious cases where content might look alright at normal scale, but will likely cause text clipping at increased scales.

Reply:
It could be displayed on a single (unclipped) line when not zoomed and need another (clipped) line when zoomed .

I agree that it is too involved without a layout system.